packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -3569,7 +3567,7 @@ struct ReduceWeightsFunctor {
 
     Kokkos::parallel_reduce(
       Kokkos::TeamThreadRange(teamMember, begin, end),
-      [=] (const size_t ii, Zoltan2_MJArrayType<array_t>& threadSum) {
+      [=] (size_t ii, Zoltan2_MJArrayType<array_t>& threadSum) {
```

Comment:
Working for OpenMP but not Cuda. It worked for Cuda until recently but some changes to Kokkos invalidated the internal handling of the array ptr. I've sent the Kokkos team some tests to demonstrate this issue. We had stopped using this for Cuda anyways due to performance issues so it was not blocking the current setup, but if we eventually did your suggestion of making the reduce path versus atomic path a parameter, then we'd need to resolve that Kokkos issue first.

Reply:
Does `Kokkos::Array` not work as an array type?  I'm a little bit uncomfortable with arrays of run-time length as reduction types, since intermediate reduction results take up CUDA shared memory.